### PR TITLE
New repo for jenkins pipeline library (groovy)

### PR DESCRIPTION
**Pull request by kharyam**
* Created a new repo (https://github.com/rhtconsulting/tssc-jenkins-library/) to host a Jenkins library that simplifies the deployment of TSSC pipelines. Jenkisfiles can simply import the library passing in a set of parameters and the underlying pipeline will be kicked off. Documentation in the README

### CODE REVIEW

File: examples/Jenkinsfile.java8 [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  remote: 'https://github.com/rhtconsulting/tssc-jenkins-library.git'
+])
+
+pipelineJava8(
+  'DEV',                 // Deployment environment defined in the tssc config file
+  'spo-git-test-km',     // Git Credentials ID
+  'spo-sonarqube-tssc',  // SonarQube Credentials ID
+  'spo-argocd-test',     // ArgoCD Credentials ID
+  'spo-artifactory-peg'  // Maven repo Credentials ID
+)
```
Comment (HunterGerlach) [inline examples/Jenkinsfile.java8]: Newline character requested

File: examples/Jenkinsfile.java8 [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  $class: 'GitSCMSource',
+  remote: 'https://github.com/rhtconsulting/tssc-jenkins-library.git'
+])
+
+pipelineJava8(
+  'DEV',                 // Deployment environment defined in the tssc config file
+  'spo-git-test-km',     // Git Credentials ID
+  'spo-sonarqube-tssc',  // SonarQube Credentials ID
+  'spo-argocd-test',     // ArgoCD Credentials ID
+  'spo-artifactory-peg'  // Maven repo Credentials ID
```
Comment (HunterGerlach) [inline examples/Jenkinsfile.java8]: What is the `-peg` reference here?

File: examples/Jenkinsfile.java8 [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+// Load the TSSC Jenkins Library master branch.
+library identifier: 'tssc@master',
+retriever: modernSCM([
+  $class: 'GitSCMSource',
+  remote: 'https://github.com/rhtconsulting/tssc-jenkins-library.git'
+])
+
+pipelineJava8(
+  'DEV',                 // Deployment environment defined in the tssc config file
+  'spo-git-test-km',     // Git Credentials ID
```
Comment (HunterGerlach) [inline examples/Jenkinsfile.java8]: What is the `-km` reference here?

File: vars/pipelineJava8.groovy [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+            } // withCredentials
+          } // container
+        } // steps
+      } // stage
+
+    } //stages
+
+  } // pipeline
+
+} //call
```
Comment (HunterGerlach) [inline vars/pipelineJava8.groovy]: Newline character needed

File: vars/pipelineJava8.groovy [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+  String JENKINS_WORKER_IMAGE_ARGOCD  = 'quay.io/tssc/tssc-tool-argocd:latest'
+  String JENKINS_WORKER_IMAGE_SKOPEO  = 'quay.io/tssc/tssc-tool-skopeo:latest'
+  String JENKINS_WORKER_IMAGE_SONAR   = 'quay.io/tssc/tssc-tool-sonar:latest'
+  String REGISTRY_SECRET_NAME         = 'quay-basic-auth'
+
+  pipeline {
+
+    agent {
+      kubernetes {
+        label "jenkins-tssc-reference-app-quarkus-rest-json-${env.BUILD_ID}"
```
Comment (HunterGerlach) [inline vars/pipelineJava8.groovy]: What do you think of parameterizing the application name to be used here and throughout, if necessary?

Comment (kharyam) [reply vars/pipelineJava8.groovy]: Good catch - I will make this a parameter

File: examples/Jenkinsfile.java8 (hunk 1 above)
Comment (kharyam) [reply examples/Jenkinsfile.java8]: Fixed

File: examples/Jenkinsfile.java8 (hunk 2 above)
Comment (kharyam) [reply examples/Jenkinsfile.java8]: developer name :) fixed

File: examples/Jenkinsfile.java8 (hunk 3 above)
Comment (kharyam) [reply examples/Jenkinsfile.java8]: Another pesky dev :)

Comment (kharyam) [reply examples/Jenkinsfile.java8]: fixed

File: vars/pipelineJava8.groovy (hunk 4 above)
Comment (kharyam) [reply vars/pipelineJava8.groovy]: fixed

File: vars/pipelineJava8.groovy (hunk 5 above)
Comment (kharyam) [reply vars/pipelineJava8.groovy]: fixed

**Review by HunterGerlach — APPROVED** [review]
Looks good to me!

**Review by jkeam — APPROVED** [review]
Looks good!

**Pull request merged**